src/category_theory/sites/sieves.lean

```diff
@@ -0,0 +1,250 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, E. W. Ayers
+-/
+
+import category_theory.over
+import category_theory.limits.shapes.finite_limits
+import category_theory.yoneda
+import order.complete_lattice
+import data.set.lattice
+
+/-!
+# Theory of sieves
+
+- For an object `X` of a category `C`, a `sieve X` is a set of morphisms to `X`
+  which is closed under left-composition.
+- The complete lattice structure on sieves is given, as well as the Galois insertion
+  given by downward-closing.
+- A `sieve X` (functorially) induces a presheaf on `C` together with a monomorphism to
+  the yoneda embedding of `X`.
+
+## Tags
+
+sieve, pullback
+-/
+
+universes v u
+namespace category_theory
+
+/--
+For an object `X` of a category `C`, a `sieve X` is a set of morphisms to `X` which is closed under
+left-composition.
+-/
+structure sieve {C : Type u} [category.{v} C] (X : C) :=
+(arrows : Π {Y}, set (Y ⟶ X))
+(downward_closed : ∀ {Y Z f} (hf : arrows f) (g : Z ⟶ Y), arrows (g ≫ f))
+attribute [simp, priority 100] sieve.downward_closed
+
+namespace sieve
+
+variables {C : Type u} [category.{v} C]
+
+variables {X Y Z : C} {S R : sieve X}
+
+/-- A sieve gives a subset of the over category of `X`. -/
+def get_arrow_set (S : sieve X) : set (over X) := λ f, S.arrows f.hom
```

Comment:
```suggestion
def set_over (S : sieve X) : set (over X) := λ f, S.arrows f.hom
```
What do you think of this name?

Reply:
Works for me - do you have a suggestion for the inverse name?